cpp/ql/lib/semmle/code/cpp/models/interfaces/FunctionInputsAndOutputs.qll

```diff
@@ -245,15 +245,18 @@ class InParameter extends FunctionInput, TInParameter {
  */
 class InParameterDeref extends FunctionInput, TInParameterDeref {
   ParameterIndex index;
+  int indirectionIndex;
 
-  InParameterDeref() { this = TInParameterDeref(index) }
+  InParameterDeref() { this = TInParameterDeref(index, indirectionIndex) }
 
   override string toString() { result = "InParameterDeref " + index.toString() }
 
   /** Gets the zero-based index of the parameter. */
   ParameterIndex getIndex() { result = index }
 
-  override predicate isParameterDeref(ParameterIndex i) { i = index }
+  override predicate isParameterDeref(ParameterIndex i, int indirection) {
+    i = index and indirectionIndex = indirection
+  }
```

Comment:
I see that for out-parameters we actually have two variants, one with and one without the extra indirection parameter. Should we have the same here?

Reply:
Yes, I think that's a great idea. I'll add that 👍

Reply:
Fixed in cd2eec0aa0c8d239af70abc17eef3abe73c1c01d.